packages/shared/wallet/src/network-selector/network-selector.tsx

```diff
@@ -134,7 +42,16 @@ export function NetworkSelector() {
 
   const handleClose = async (chain?: Chain) => {
     if (chain) {
-      switchNetwork(parseInt(chain.id));
+      const currentChain =
```

Comment:
There should not be a need to do this parsing if the component uses the `Network` enum rather than some other enum, so you can add a property to the `chains` array which is the `Network` enum value.

Reply:
We can't add properties to chains because it breaks onboard. But we can use the label as this property